modules/irsa/main.tf

```
@@ -17,7 +17,7 @@ resource "kubernetes_service_account_v1" "irsa" {
   count = var.create_kubernetes_service_account ? 1 : 0
   metadata {
     name        = var.kubernetes_service_account
-    namespace   = var.kubernetes_namespace
+    namespace   = try(kubernetes_namespace_v1.irsa[count.index].metadata[0].name, var.kubernetes_namespace)
```

we don't tie it to the count because they aren't associated. if namespace creation is enabled, there isn't a 2nd index (position `[1]`):

```suggestion
    namespace   = try(kubernetes_namespace_v1.irsa[0].metadata[0].name, var.kubernetes_namespace)
```